src/islpr.cc

```diff
@@ -0,0 +1,401 @@
+  isl_printer * ip = NULL;
+  ip = isl_printer_to_str(ctx);
+  
+  // load Relation r into ISL map
+  isl_map* imap = NULL;
+  imap = isl_map_read_from_str(ctx, relstr.c_str());
+
+  // get string back from ISL map
+  char * cstr;
+  isl_printer_set_output_format(ip , ISL_FORMAT_ISL);
+  isl_printer_print_map(ip ,imap);
+  cstr=isl_printer_get_str(ip);
+  string stringFromISL = cstr;
+  
+  // clean-up
+  isl_printer_flush(ip);
+  isl_printer_free(ip);
+  free(cstr);
+  isl_map_free(imap);
+  imap= NULL;
+  isl_ctx_free(ctx); 
+
+  return stringFromISL;
+}
+
+string project_out (string relstr, unsigned pos)
+{
+  // Get an isl context
+  isl_ctx *ctx;
+  ctx = isl_ctx_alloc();
+  
+  // Get an isl printer and associate to an isl context
+  isl_printer * ip = NULL;
+  ip = isl_printer_to_str(ctx);
+  
+  // load Relation r into ISL map
+  isl_map* imap = NULL;
+  imap = isl_map_read_from_str(ctx, relstr.c_str());
+
+  imap = isl_map_project_out(imap, isl_dim_in, pos, 1);
+
+  // get string back from ISL map
+  char * cstr;
+  isl_printer_set_output_format(ip , ISL_FORMAT_ISL);
+  isl_printer_print_map(ip ,imap);
+  cstr=isl_printer_get_str(ip);
+  string stringFromISL = cstr;
+  
+  // clean-up
+  isl_printer_flush(ip);
+  isl_printer_free(ip);
+  free(cstr);
+  isl_map_free(imap);
+  imap= NULL;
+  isl_ctx_free(ctx); 
+
+  return stringFromISL;
+}
+
+std::string set_project_out (std::string relstr, unsigned pos)
+{
+  // Get an isl context
+  isl_ctx *ctx;
+  ctx = isl_ctx_alloc();
+  
+  // Get an isl printer and associate to an isl context
+  isl_printer * ip = NULL;
+  ip = isl_printer_to_str(ctx);
+  
+  // load Relation r into ISL map
+  isl_set* iset = NULL;
+  iset = isl_set_read_from_str(ctx, relstr.c_str());
+
+  iset = isl_set_project_out(iset, isl_dim_out, pos, 1);
+
+  // get string back from ISL map
+  char * cstr;
+  isl_printer_set_output_format(ip , ISL_FORMAT_ISL);
+  isl_printer_print_set(ip ,iset);
+  cstr=isl_printer_get_str(ip);
+  std::string stringFromISL = cstr;
+  
+  // clean-up
+  isl_printer_flush(ip);
+  isl_printer_free(ip);
+  free(cstr);
+  isl_set_free(iset);
+  iset= NULL;
+  isl_ctx_free(ctx); 
+
+  return stringFromISL;
+}
+
+
+// Takes an UFS and returns a variable name representing it:
+// col(j) -> col_j;     idx(ip+1) -> idx_ipP1
+string getUF_varname (string ufs)
+{
+  char * cstr = new char [ufs.length()+1];
+  std::strcpy (cstr, ufs.c_str());
+
+  for(int i = 0 ; i < ufs.length() ; i++ )
+  {
+     if (cstr[i] == '(')  cstr[i] = '_';
+     else if (cstr[i] == '+')  cstr[i] = 'P';
+     else if (cstr[i] == '-')  cstr[i] = 'M';
+  }
+  cstr[ufs.length()-1] = cstr[ufs.length()];
+
+  string res(cstr);
+  return res;
+}
+
+int main(int argc, char **argv)
+{
+
+  string GS_F("[n] -> { [i] -> [ip] : i < ip and"
+               " exists j,jp: 0 <= i,ip <= n and 0 <= j,jp <= n }");
+  string isl_str_F = getRelationStringFromISL(GS_F);  
+
+  cout<<endl;
+  cout<<isl_str_F;
+  cout<<endl;
+/*
```

Comment:
Everything below here is all comments.  You don't want to include that in the merge.